kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -359,6 +359,9 @@ protected void handleTopicMetadataRequest(KafkaHeaderAndRequest metadataHar,
         CompletableFuture<Map<String, List<TopicName>>> pulsarTopicsFuture = new CompletableFuture<>();
 
         if (topics == null || topics.isEmpty()) {
+            // clean all cache when get all metadata for the first metadata request.
```

Comment:
The official Kafka clients won't request all metadata during produce and consume. I think the old version librdkafka would request all metadata. If I remember correctly, librdkafka also dosn't request all metadata from 1.0.x. So this behavior may affect other clients.

Reply:
What you say is right,  there is no request for all metadata for librdkafka(1.2.2) and java client(2.5.1)  I tested.

Reply:
the latest commit has updated:
1,  keep clean all cache when getting all metadata for librdkafka(<1.0.0)
2,  remove kop address after channel close.